libs/core/algorithms/include/hpx/parallel/datapar/find.hpp

```diff
@@ -67,22 +68,23 @@ namespace hpx { namespace parallel { inline namespace v1 { namespace detail {
     };
 
     template <typename ExPolicy, typename Iterator, typename Sentinel,
-        typename T, typename Proj = util::projection_identity>
-    inline constexpr std::enable_if_t<
-        hpx::is_vectorpack_execution_policy<ExPolicy>::value, Iterator>
-    tag_invoke(sequential_find_t<ExPolicy>, Iterator first, Sentinel last,
+        typename T, typename Proj = util::projection_identity,
+        HPX_CONCEPT_REQUIRES_(
+            hpx::is_vectorpack_execution_policy<ExPolicy>::value)>
+    HPX_HOST_DEVICE HPX_FORCEINLINE Iterator tag_invoke(
+        sequential_find_t<ExPolicy>, Iterator first, Sentinel last,
         T const& val, Proj proj = Proj())
     {
         return datapar_find<ExPolicy>::call(first, last, val, proj);
     }
 
     template <typename ExPolicy, typename FwdIter, typename Token, typename T,
-        typename Proj>
-    inline constexpr std::enable_if_t<
-        hpx::is_vectorpack_execution_policy<ExPolicy>::value, void>
-    tag_invoke(sequential_find_t<ExPolicy>, std::size_t base_idx,
-        FwdIter part_begin, std::size_t part_count, Token& tok, T const& val,
-        Proj&& proj)
+        typename Proj,
+        HPX_CONCEPT_REQUIRES_(
+            hpx::is_vectorpack_execution_policy<ExPolicy>::value)>
+    HPX_HOST_DEVICE HPX_FORCEINLINE void tag_invoke(sequential_find_t<ExPolicy>,
+        std::size_t base_idx, FwdIter part_begin, std::size_t part_count,
+        Token& tok, T const& val, Proj&& proj)
     {
         return datapar_find<ExPolicy>::call(base_idx, part_begin, part_count,
```

Comment:
```suggestion
        return datapar_find<std::decay_t<ExPolicy>>::call(base_idx, part_begin, part_count,
```
?

Reply:
@hkaiser Is `decay`ing ExPolicy in all the `tag_invoke`s required :) ?
Because, we however `decay` it in the CPOs before using the `ExPolicy`. 

**For example :** 
We decay in the CPOs here
https://github.com/srinivasyadav18/hpx/blob/simd_find/libs/core/algorithms/include/hpx/parallel/datapar/find.hpp#L40
https://github.com/srinivasyadav18/hpx/blob/2024bd2318c368bd4c236c629a16037b519b6c2d/libs/core/algorithms/include/hpx/parallel/datapar/find.hpp#L61

For this `tag_invoke`:
https://github.com/srinivasyadav18/hpx/blob/simd_find/libs/core/algorithms/include/hpx/parallel/datapar/find.hpp#L72-L93

Reply:
We don't need to do it if we know it's already decayed. Good catch.